internal/web/server.go

```diff
@@ -0,0 +1,179 @@
+package web
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"html/template"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/styrainc/regal/internal/explorer"
+	"github.com/styrainc/regal/internal/lsp/cache"
+	"github.com/styrainc/regal/internal/lsp/clients"
+)
+
+//go:embed assets/*
+var assets embed.FS
+
+//nolint:gochecknoglobals,lll
+var tpl = template.Must(template.New("main").Parse(`
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta charset="utf-8"/>
+		<link rel="stylesheet" href="/assets/missing.min.css">
+		<script src="https://unpkg.com/htmx.org@1.8.4" integrity="sha384-wg5Y/JwF7VxGk4zLsJEcAojRtlVp1FKKdGy1qN+OMtdq72WRvX/EdRdqg/LOhYeV" crossorigin="anonymous"></script>
+	<body>
+		<header>
+			<h1>OPA Explorer<v-h>:</v-h><sub-title>Inspect compiler stages</sub-title></h1>
+			<nav>
+				<p class="tool-bar">
+					<img src="https://openpolicyagent.org/badge/v0.68.0"/>
+					<a href="https://github.com/srenatus/opa-explorer">GitHub</a>
+				</p>
+			</nav>
+		</header>
```

Comment:
Does the header still make sense? I'm not sure. I'd either slim it down to something that matches the source action description, or remove it.